src/realtimecontroller/DefaultRealtimeController.ts

```diff
@@ -267,12 +267,26 @@ export default class DefaultRealtimeController implements RealtimeController {
       this.onError(e);
     }
   }
-
-  realtimeUnsubscribeFromVolumeIndicator(attendeeId: string): void {
-    try {
-      delete this.state.volumeIndicatorCallbacks[attendeeId];
-    } catch (e) {
-      this.onError(e);
+  realtimeUnsubscribeFromVolumeIndicator(
+    attendeeId: string,
+    callback?: (
+      attendeeId: string,
+      volume: number | null,
+      muted: boolean | null,
+      signalStrength: number | null,
+      externalUserId?: string
+    ) => void
+  ): void {
+    if (!callback) {
+      try {
+        delete this.state.volumeIndicatorCallbacks[attendeeId];
+      } catch (e) {
+        this.onError(e);
+      }
+    } else {
+      if(this.state.volumeIndicatorCallbacks[attendeeId]) {
+        this.state.volumeIndicatorCallbacks[attendeeId] = this.state.volumeIndicatorCallbacks[attendeeId].filter(f => f !== callback);
```

Comment:
You should use `splice` here. See how it's done on line 366.

Reply:
(Yes, strictly speaking `filter` will catch the cases where you subscribe multiple times with the same callback and want it removed in one call. `splice` is the approach we've taken elsewhere for symmetry, so N subscribes need N removals.)